### DISCUSSION TITLE
Issues upgrading to Version 0.5.3

### DISCUSSION THREAD

**Discussion by k2skaterii** [guydavis/machinaris#252] [Q&A] [answered]
*** Kudos to Guy for updating machinaris with the latest version of Chia as fast as he did!  Thank you! ***

I just attempted to upgrade my plotting instances to V 0.5.3... I started with my plotters and have not upgraded my node or my harvester yet.  

After upgrading my plotters, none of them showed up in my worker's page.  The apisrv.log (attached) indicates that it's a permissions problem, and even includes the command to fix the permissions problem.  

It also indicates the version of chia-blockchain is 1.2.5.dev0 not 1.2.4 - I've seen similar behavior when installing chia on a FreeBSD system, so it may not be an issue.  Worth noting either way.

```
One or more SSL files were found with permission issues.
Run `chia init --fix-ssl-permissions` to fix issues.
1.2.5.dev0

```

Before connecting to each system and running the command to fix the permissions, I wanted to mention it here in case these errors are symptoms of a larger problem.

[apisrv.log](https://github.com/guydavis/machinaris/files/7071090/apisrv.log)

**Comment by k2skaterii** [comment, marked as answer]
Quick update.  I went ahead and ran the `chia init --fix-ssl-permissions` as suggested in the apisrv.log.  That seems to have fixed the problem. 

Since that command fixed the problem on my plotters, I went ahead and updated my harvester and node.  I did not experience similar issues when upgrading my node or my harvester.  Which leads me to believe the error is related to a change in chia Version 1.2.4 and the fact that I never did import my keys on my plotters.  

AND... I just noticed that Chia v1.2.5 was released 19 hours ago.  GREAT!

**Reply by guydavis** [reply]
Thanks for the report.  Yes, I did ensure all the other modes (fullnode, harvester) applied the correct SSL file permissions, but I neglected that a plotter-only system might hit this issue when using the Chia plotter.  I'll fix this straight-away in the next release, including Chia 1.2.5.  Thanks!

**Comment by PMiani** [comment]
I also had issues with all my workers. I did not have the mentioned error in the `apisrv.log` or `webui.log`, but was getting HTTP 500.
I can confirm `chia init --fix-ssl-permissions` solve the issue. However the plot "Partial Proofs to Pools" in the home page only show last hour.. on refresh only last hour is shown.

**Reply by PMiani** [reply]
Furthermore, any workers replies with a HTTP 500 on endpoint `http://<fullnode>/worker?hostname=<worker>` (when you click on a worker in the Workers page). Also, plotting logs cannot be seen at `http://<fullnode>/logs?log=plotting&hostname=<fullnode>&log_id=,id>` as it shows `No log file found!`. Logs from workers can be viewed as usual.

**Reply by k2skaterii** [reply]
@PMiani - Nice that `chia init --fix-ssl-permissions` fixed your issues on your workers even though you didn't have an error in `apisrv.log`.  Curious that it wasn't in the log, were you looking at the logs on your controller or on the workers themselves?  

I noticed the same thing with the Partial Proofs only going back an hour and I've seen the same issue clicking on my worker names from the workers page.  New features not working 100% - either way they are pretty cool additions!

**Reply by PMiani** [reply]
@k2skaterii indeed, the error was in the container log and not in the *.log files of the workers. In fact the web server reported its status as `started`. Fullnode fixed itself as mentioned by @guydavis. I myself run in TrueNAS Scale, but don't use the charts from TrueNAS. Also, unless you have a RAID 0 for plotting and much and fast RAM, I highly recommend not to plot in ZFS pools. I use a single SSD mapped directly to the plotter container. My plotting time went down substantially.

**Reply by k2skaterii** [reply]
> I myself run in TrueNAS Scale, but don't use the charts from TrueNAS. Also, unless you have a RAID 0 for plotting and much and fast RAM, I highly recommend not to plot in ZFS pools. 

If you are not using the TrueNAS charts on TrueNAS Scale - How did you get it setup.  I just replaced TrueNAS-core with scale the other day, haven't had a chance to do much with charts or other containers - intentionally avoided setting up a dedicated VM.  How you set machinaris up in TrueNAS Scale might be worthy of it's own discussion post so others can find it easilly.

As far as the storage I'm dedicating to plotting - I've got some decisions to make related to my storage setup.  While it might be sub-optimal, right now I'm plotting on an existing 14 disk ZFS pool, 14x 300GB 10k SAS drives.  I did tweak some things when I created the plotting dataset, disabling compression and setting "Enable Atime" to OFF. 

I almost forgot that I used my TrueNAS plotter and produced 1x K32 with the MadMax plotter - **171 Minutes** - 2 threads, 256 buckets, 2 threads on the rmulti2 setting.  Not great, but also not bad.  I've got other VMWare VM's plotting with MadMax (all disk, no RAM disk) which are creating K32's in 90 minutes.  I've got 2 of those VMs running at the same time - 3TB/day.  Thinking that the TrueNAS plotter will be used to spit out K33's & K34's.
